controller/src/main/java/org/jboss/as/controller/AbstractAddStepHandler.java

```diff
@@ -391,27 +380,29 @@ public Resource createResource(OperationContext context, ModelNode operation) {
     }
 
     public static class Parameters {
-        protected Set<AttributeDefinition> attributes = null;
+        protected Collection<AttributeDefinition> attributes = List.of();
```

Comment:
@pferraro This is a breaking change, so why not go for it and make this private as well?

Reply:
Ugh - I missed that this was protected instead of private...  I did not intend for this to introduce breaking changes.
Frankly, this PR is mostly obsoleted by WFCORE-6407, which proposes to deprecate the AbstractAddStepHandler.Parameters class, as well as all non-default constructors of AbstractAddStepHandler and AbstractWriteAttributeHandler.

Reply:
Fixed.